src/TesApi.Web/Storage/TerraStorageAccessProvider.cs

```diff
@@ -0,0 +1,161 @@
+            ArgumentNullException.ThrowIfNull(terraOptions);
+
+            this.terraOptions = terraOptions.Value;
+        }
+
+        /// <inheritdoc />
+        public override Task<bool> IsPublicHttpUrlAsync(string uriString)
+        {
+            var isHttpUrl = TryParseHttpUrlFromInput(uriString, out var uri);
+
+            if (!isHttpUrl)
+            {
+                return Task.FromResult(false);
+            }
+
+            if (HttpUtility.ParseQueryString(uri.Query).Get("sig") is not null)
+            {
+                return Task.FromResult(true);
+            }
+
+            if (StorageAccountUrlSegments.TryCreate(uriString, out var parts))
+            {
+                if (IsTerraWorkspaceContainer(parts.ContainerName) && IsTerraWorkspaceStorageAccount(parts.AccountName))
+                {
+                    return Task.FromResult(false);
+                }
+            }
+
+            return Task.FromResult(true);
+        }
+
+        /// <inheritdoc />
+        public override async Task<string> MapLocalPathToSasUrlAsync(string path, bool getContainerSas = false)
+        {
+            ArgumentException.ThrowIfNullOrEmpty(path);
+
+            var normalizedPath = path.TrimStart('/');
+
+            if (getContainerSas)
+            {
+                return await GetMappedSasContainerUrlFromWsmAsync(normalizedPath);
+            }
+
+            if (IsItKnownExecutionFilePath(normalizedPath))
+            {
+                return await GetMappedSasUrlFromWsmAsync(normalizedPath);
+            }
+
+            if (!StorageAccountUrlSegments.TryCreate(normalizedPath, out var segments))
+            {
+                throw new Exception(
+                    "Invalid path provided. The path must be a valid blob storage url or a path with the following format: /accountName/container");
+            }
+
+            CheckIfAccountAndContainerAreWorkspaceStorage(segments.AccountName, segments.ContainerName);
+
+            return await GetMappedSasUrlFromWsmAsync(segments.BlobName);
+        }
+
+        private async Task<string> GetMappedSasContainerUrlFromWsmAsync(string pathToAppend)
+        {
+            //an empty blob name gets a container Sas token
+            var tokenInfo = await GetSasTokenFromWsmAsync(CreateTokenParamsFromOptions(blobName: string.Empty, SasContainerPermissions));
+
+            var urlBuilder = new UriBuilder(tokenInfo.Url);
+
+            urlBuilder.Path += pathToAppend;
+
+            return urlBuilder.ToString();
+        }
+
+        private async Task<string> GetMappedSasUrlFromWsmAsync(string blobName)
+        {
+            var tokenParams = CreateTokenParamsFromOptions(blobName, SasBlobPermissions);
+
+            var tokenInfo = await GetSasTokenFromWsmAsync(tokenParams);
+
+            logger.LogInformation($"Successfully obtained the Sas Url from Terra. Requested blobName:{blobName}. Wsm resource id:{terraOptions.WorkspaceStorageContainerResourceId}");
```

Comment:
The blobName could contain customer information in the filename.  Usually it is something like a sample ID, but still would be better not to log it

Reply:
Makes sense. Will fix in my next PR.